pkgs/os-specific/linux/roccat-tools/default.nix

```diff
@@ -0,0 +1,112 @@
+{ stdenv, fetchurl, cmake, dbus, glib, libcanberra, atk, cairo
+, gdk_pixbuf, gtk2, pango, libnotify, libgudev, xorg, sqlite
+, gegl, dbus_glib, gettext, pkgconfig, shadow, devices ? false
+}:
+
+# Use a semi-colon separated list to specify devices in configuration.nix eg -
+# nixpkgs.config.roccat-tools.devices = "konepuremilitary;kiro";
+# Supported devices are : arvo;isku;iskufx;koneplus;konepure;konepuremilitary
+# ;konepureoptical;konextd;konextdoptical;kovaplus;lua;nyth;pyra;ryosmk;ryostkl;savu
+# ;tyon;kone;kiro;kova2016
+# By default all devices are built.
+
+stdenv.mkDerivation rec {
+  version = "3.9.0";
+  name = "roccat-tools-${version}";
+  src = fetchurl {
+    url = "mirror://sourceforge/libgaminggear/libgaminggear-0.11.2.tar.bz2";
+    sha256 = "99abcb627242aa96b597ed831fd1e6df1e8a3958fbaf6cc8e8fdf04d62570d0e";
+  };
+
+  toolsSrc = fetchurl {
+    url = "mirror://sourceforge/roccat/${name}.tar.bz2";
+    sha256 = "ea90b53cf4707c76a41d0f67711eca91e21ef41be253ab45b113a55ac1f278c1";
+  };
+
+  dontUseCmakeBuildDir=true;
+
+  nativeBuildInputs = [ cmake pkgconfig ];
+
+  buildInputs = [ stdenv cmake dbus glib libcanberra atk cairo gdk_pixbuf 
+		  gtk2 pango libnotify libgudev xorg.libX11 
+		  xorg.libpthreadstubs xorg.libXdmcp 
+		  xorg.libxshmfence sqlite gegl 
+		  dbus_glib gettext pkgconfig shadow ];
+
+  config1 = [
+    "-DINSTALL_LIBDIR=lib"
+    "-DINSTALL_CMAKE_MODULESDIR=usr/share/cmake/Modules"
+    "-DINSTALL_PKGCONFIGDIR=usr/share/pkgconfig"
+    "-DWITH_UHID=TRUE"
+    "-DWITH_DOC=FALSE"
+    "-DGFX_PLUGIN_DIR=lib/gaminggear_plugins"
+    "-DCMAKE_MODULE_PATH=usr/share/cmake/Modules"
+  ];
+
+  config2 = if builtins.isString (devices) then [
+    "-DDEVICES=${devices}"
+  ] else [
+    ""
+  ];
+
+  # We build both libgaminggear and roccat-tools in one derivation
+  # because roccat-tools writes into libgaminggear's plugin directory.
+
+  builder = builtins.toFile "builder.sh"
```

Comment:
@heydojo Well, I do have read the code and I also don't understand what's the problem here. Could you please elaborate instead of reiterating that we should read the code? That would be the real adult thing to do instead of refusing to consider code changes without any justification.

My guess here: libgaminggear and roccat-tools should probably be two separate derivations.